bash/blddist

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# blddist: Script to create z390 distribution zip file
+# passing parm "clean" will remove repo ignored files and rebuild prior to dist build
+
+set -e                          # exit on all errors
+cd $(dirname $0)                # set to directory of script
+cd ..                           # cd back to root       
+
+root_dir=$(pwd)
+version_tag=$(git describe)
+
+if [ "$1" == "clean" ];         # if "clean" parm passed
+then
+    git clean -dfX              # delete all git ignored files
+fi
+
+mkdir -p build                  # create build folder if does not exist
+rm -rf build/z390_${version_tag}    # delete version build directory
+
+bash/bldjar                     # create the z390 jar
+
+cd build
+mkdir z390_${version_tag}
+cd z390_${version_tag}
+
+# add core files to the build dir
+cp ${root_dir}/z390.jar .       # jar
+core_dirs=( bash bat mac zcobol cics vse linklib )
+for dir in ${core_dirs[@]}; do
+    cp -R ${root_dir}/${dir} .        # unix bash scripts
+done
+
+# add test and supplimental components
+supp_dirs=( assist demo guam qsam rt soa sort structuredmacros tests vsam1 vsam2 zopcheck zpar )
+for dir in ${supp_dirs[@]}; do
+    cp -R ${root_dir}/${dir} .        # unix bash scripts
+done
+
+# create zip package
+cd ..
+zip -r z390_${version_tag}.zip z390_${version_tag}
```

Comment:
Very nice.

I see that blddist is invoked from build.sh. No parm(s) passed to blddist. Is it invoked anywhere else?

I still object to the use of git commands in the build process (so far "git clean -dfX" and "git describe"), but do not see an easy way to avoid them.

The comments

Reply:
Thanks. It's the power of Google 😁
I understand your reservations. 
If you have a better approach let me know and I will see what I can do.

Reply:
I don't. Let's both think about it for a few days. If nothing jumps out, I'll approve this request so we can move on.

Reply:
Sounds like a plan.. I do like the versions being driven by tags and commits in the repo.. just makes it clear which version of the source the jar was built with - remember, its not just the version number but the number of commits and the hash if its a build between versions.
Each commit could look like this:  v1.7.07 -> v1.7.07-1-abcdef1 -> v1.7.07-2-1fedcba -> v1.7.08
Without git, how can you work this out? There may be a way, I just can't think of how without some artefact from the source control system.

Reply:
Where is the distributed "source" built?

Reply:
In the releases view, it always provides a link to a zipped up version of the repo for the specific release. 
See https://github.com/adelosa/z390/releases/tag/v1.7.08 for example.
You don't have to do anything explicitly for this to be created. Its just available.
![image](https://user-images.githubusercontent.com/8604267/137688409-0cc34c7b-af46-4117-99af-8f8632300a2f.png)